draft-aanchal-time-implementation-guidance.md

```diff
@@ -0,0 +1,358 @@
+values that govern the length of time for which an object is valid or usable. 
+
+Since relative time does not express a point in time, it does not rely on
+synchronized clocks between systems but only on a shared clock rate.
+
+# Keeping Time: Different Clocks
+
+In this section, we will have a look at the different clocks a system uses and
+how it maintains these clocks
+
+## Native Clock
+
+Each system has its own perception of time.  It gains access it via its native
+clock.  Typcially, this clock counts cycles of an oscillator but some systems
+use process CPU times or thread CPU timers (via timers provided by the CPU).
+The quality of the native clock therefore dependends on either the stability of
+the oscillator or the CPU timer.
+
+The timescale of the native clock is a purely subjective -- no general meaning
+can be attached to any specific clock value.  One can only obtain relative time
+by comparing two values.  Because the value of the native clock always grows at
+a steady pace, never decreases, never make unexpected jumps, and never skips,
+the difference between two clock values provides the time intervall between the
+two measurements.
+
+The independence of the native clock from any external time sources renders it
+resistant to any manipulation but in return there is no guarantee that its
+clock rate is similar to that of any other system.  This difference in rate,
+especially when compared to a reference clock, is called clock drift.
+
+Clock drift depends on the quality of the clock itself but also on factors such
+as system load or ambient temperatur which makes it hard to predict.
+
+## World Clock
+
+The native clock only provides means to measure relative time.  In order to be
+able to also process absolute time, it needs to be synchronized with a global
+reference clock.  Since this clock strives to be the same on all systems, we
+call it the world clock.
+
+There are a number of ways to maintain the world clock based on the system's
+native clock.
+
+  * The first is to manually maintain an offset between values of the native
+    clock and the reference world clock.  Because of the clock drift of the
+    native clock, this offset needs to be updated from time to time if a
+    minimal divergence from the reference clock is to be maintained.
+
+  * Secondly, a hardware clock provided by the system and set to be equivalent
+    to the reference time can be used, allowing the system to retain the offset
+    across reboots.
+
+  * Finally, the reference clock can be obtained from an external time source.
+    Typically, the Internet is used through a variety of timing protocols
+    including the Network Time Protocol2 (NTP), Chrony, SNTP, OpenNTP and
+    others.
+
+Each of these approaches has own problems attached to it.
+
+  * Manual configurations can be subject to errors and misconfiguration.
+    Also, for mobile devices, when moving between time zones, the offset
```

Comment:
Is that really worth mentioning? Aren’t time zones applied only when displaying time while the actual time kept is UTC?

Reply:
Agree, I'd like to remove that line about mobile devices too.